src/test/kotlin/io/github/mojira/arisa/modules/CrashModuleTest.kt

```diff
@@ -0,0 +1,600 @@
+
+        result.shouldBeRight(ModuleResponse)
+        resolvedAsDupe.shouldBeTrue()
+    }
+
+    "should resolve as dupe when the configured crash is a java crash" {
+        var resolvedAsDupe = false
+
+        val module = CrashModule(
+            { Unit.right() },
+            { resolvedAsDupe = true; Unit.right() },
+            { Unit.right() },
+            { Unit.right() },
+            { Unit.right() },
+            listOf("txt"),
+            listOf(CrashDupeConfig("java", "ig75icd64\\.dll", "MC-32606")),
+            10
+        )
+        val request = CrashModuleRequest(emptyList(), JAVA_CRASH, Calendar.getInstance().time)
+
+        val result = module(request)
+
+        result.shouldBeRight(ModuleResponse)
+        resolvedAsDupe.shouldBeTrue()
+    }
+
+    "should resolve as dupe when the configured crash uses regex" {
+        var resolvedAsDupe = false
+
+        val module = CrashModule(
+            { Unit.right() },
+            { resolvedAsDupe = true; Unit.right() },
+            { Unit.right() },
+            { Unit.right() },
+            { Unit.right() },
+            listOf("txt"),
+            listOf(CrashDupeConfig("java", "ig[0-9]{1,2}icd[0-9]{2}\\.dll", "MC-32606")),
+            10
+        )
+        val request = CrashModuleRequest(emptyList(), JAVA_CRASH, Calendar.getInstance().time)
+
+        val result = module(request)
+
+        result.shouldBeRight(ModuleResponse)
+        resolvedAsDupe.shouldBeTrue()
+    }
+
+    "should link to configured ticket when resolving as dupe" {
+        val module = CrashModule(
+            { Unit.right() },
+            { Unit.right() },
+            { it.shouldBe("MC-297").right() },
+            { Unit.right() },
+            { Unit.right() },
+            listOf("txt"),
+            listOf(CrashDupeConfig("minecraft", "Pixel format not accelerated", "MC-297")),
+            10
+        )
+        val request = CrashModuleRequest(emptyList(), EXAMPLE_CRASH, Calendar.getInstance().time)
+
+        val result = module(request)
+
+        result.shouldBeRight(ModuleResponse)
+    }
+
+    "should prefer crash that is not modded, if modded crash appears first" {
+        var resolvedAsDupe = false
+
+        val module = CrashModule(
+            { Unit.right() },
+            { resolvedAsDupe = true; Unit.right() },
+            { Unit.right() },
+            { Unit.right() },
+            { Unit.right() },
+            listOf("txt"),
+            listOf(CrashDupeConfig("minecraft", "Pixel format not accelerated", "MC-297")),
+            10
+        )
+        val modded = mockAttachment("crash_modded.txt", Calendar.getInstance().time, EXAMPLE_CRASH.toByteArray())
+        val dupe = mockAttachment("crash_dupe.txt", Calendar.getInstance().time, EXAMPLE_CRASH.toByteArray())
+        val request = CrashModuleRequest(listOf(modded, dupe), "", Calendar.getInstance().time)
+
+        val result = module(request)
+
+        result.shouldBeRight(ModuleResponse)
```

Comment:
Minor, you can do result shouldBeRgiht ModuleResponse (spaces instead of . and () )

Reply:
I just copied it from other classes, that's how it's done everywhere else in the project :D